Add GH workflow to build and release on GH and PyPI (#1946)
theupdateframework/python-tuf: docs/RELEASE.md

# Release process

* Ensure you have a backup of all working files and then remove files not tracked by git
  `git clean -xdf`. **NOTE**: this will delete all files in the tuf tree that aren't

>   tracked by git

Reviewer: removing this seems right, but there is a related issue: https://github.com/theupdateframework/python-tuf/issues/1942 (build results for verify_release are still affected by randome files in tree).

I can have a look at the workaround (making a clean checkout before running build).

Reviewer: made #1947 for this

Author: Thanks, I considered leaving that line, but secretly hoped that we could find a different fix (`git clean -xdf` already killed my pre commit hook once). https://github.com/theupdateframework/python-tuf/pull/1947 looks great for this.

Reviewer: 1947 is merged, it's all good now I think :crossed_fingers:

3. Create a PR with updated `CHANGELOG.md` and version bumps

&#10132; Review PR on GitHub

4. Once the PR is merged, pull the updated `develop` branch locally
5. Create a signed tag for the version number on the merge commit
  `git tag --sign vA.B.C -m "vA.B.C"`
6. Push the tag to GitHub `git push origin vA.B.C`

  *A push triggers the [CI workflow](.github/workfows/ci.yml), which, on success, triggers
  the [CD worfklow](.github/workfows/cd.yml), which builds source dist and wheel,
  creates a preliminary GitHub release under `vA.B.C-rc`, and pauses for review.*

7. Run `verify_release --skip-pypi` locally to make sure a build on your machine matches
  the preliminary release artifacts published on GitHub.

> &#10132; [Review *deployemnt*](https://docs.github.com/en/actions/managing-workflow-runs/reviewing-deployments) on GitHub

Reviewer: ```suggestion
&#10132; [Review *deployment*](https://docs.github.com/en/actions/managing-workflow-runs/reviewing-deployments) on GitHub
```

>   *An approval resumes the CD workflow to publish the release on PyPI, and to finalize the
>   GitHub release (removse `-rc` suffix and updates release notes).*

Reviewer: ```suggestion
  *Approval resumes the CD workflow to publish the release on PyPI, and to finalize the
  GitHub release (removes `-rc` suffix and updates release notes).*
```

> 8. `verify_release` may be used again to make sure the release artifacts PyPI.

Reviewer: ```suggestion
8. `verify_release` may be used again to make sure the PyPI release artifacts match local build as well.
```

Author: 🤦 Apologies!
